lib/versioner/simple.go

```diff
@@ -73,12 +72,14 @@ func (v Simple) Archive(filePath string) error {
 	// Use all the found filenames. "~" sorts after "." so all old pattern
 	// files will be deleted before any new, which is as it should be.
 	versions := util.UniqueTrimmedStrings(append(oldVersions, newVersions...))
-	sort.Strings(versions)
 
-	if len(versions) > v.keep {
-		for _, toRemove := range versions[:len(versions)-v.keep] {
+	// Versions with mtimes
+	versionsWithMtimes := versionsToVersionsWithMtime(v.versionsFs, versions)
+
+	if len(versionsWithMtimes) > v.keep {
+		for _, toRemove := range versionsWithMtimes[:len(versionsWithMtimes)-v.keep] {
```

Comment:
The list of versions must be properly sorted here, which they are not (they are sorted by name or whatever glob happened to return rather than mtime). I think `versionsToVersionsWithMtime` should do this sorting so the caller doesn't need to remember.